master broken on Android

```
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp: In member function 'bool {anonymous}::IncrementalChangeInfo::advance_incremental(realm::VersionID)':
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:428:45: error: no matching function for call to 'std::vector<realm::_impl::TransactionChangeInfo>::push_back(<brace-enclosed initializer list>)'
                 std::move(m_current->lists)});
                                             ^
/Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:428:45: note: candidates are:
In file included from /usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/vector:64:0,
                 from /Users/cm/Realm/realm-object-store/src/schema.hpp:23,
                 from /Users/cm/Realm/realm-object-store/src/shared_realm.hpp:22,
                 from /Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.hpp:22,
                 from /Users/cm/Realm/realm-object-store/src/impl/realm_coordinator.cpp:19:
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1038:7: note: void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = realm::_impl::TransactionChangeInfo; _Alloc = std::allocator<realm::_impl::TransactionChangeInfo>; std::vector<_Tp, _Alloc>::value_type = realm::_impl::TransactionChangeInfo]
       push_back(const value_type& __x)
       ^
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1038:7: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'const value_type& {aka const realm::_impl::TransactionChangeInfo&}'
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1057:7: note: void std::vector<_Tp, _Alloc>::push_back(std::vector<_Tp, _Alloc>::value_type&&) [with _Tp = realm::_impl::TransactionChangeInfo; _Alloc = std::allocator<realm::_impl::TransactionChangeInfo>; std::vector<_Tp, _Alloc>::value_type = realm::_impl::TransactionChangeInfo]
       push_back(value_type&& __x)
       ^
/usr/local/Cellar/android-ndk-r10e/r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/bits/stl_vector.h:1057:7: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'std::vector<realm::_impl::TransactionChangeInfo>::value_type&& {aka realm::_impl::TransactionChangeInfo&&}'
make[2]: *** [src/CMakeFiles/realm-object-store.dir/impl/realm_coordinator.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/realm-object-store.dir/all] Error 2
make: *** [all] Error 2
```

Reproduce:
Follow the instructions for building for Android. `make` will crash with the above.

--- a/src/impl/collection_notifier.hpp
+++ b/src/impl/collection_notifier.hpp
@@ -49,6 +49,16 @@ struct TransactionChangeInfo {
     std::vector<ListChangeInfo> lists;
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
+
+    // Required by GCC 4.9
+    TransactionChangeInfo() {}
+    TransactionChangeInfo(std::vector<bool> table_mods,
+                          std::vector<bool> table_moves,
+                          std::vector<ListChangeInfo> ls)
+            : table_modifications_needed(table_mods),
+              table_moves_needed(table_moves),
+              lists(ls)
+    {}
 };
 
 class DeepChangeChecker {

--- a/src/impl/collection_notifier.hpp
+++ b/src/impl/collection_notifier.hpp
@@ -50,7 +50,8 @@ struct TransactionChangeInfo {
     std::vector<CollectionChangeBuilder> tables;
     bool track_all = false;
 
-    // Required by GCC 4.9
+#if __GNUC__ < 5
+    // GCC 4.9 does not support C++14 braced-init with NSDMIs
     TransactionChangeInfo() {}
     TransactionChangeInfo(std::vector<bool> table_modifications_needed,
                           std::vector<bool> table_moves_needed,
@@ -59,6 +60,7 @@ struct TransactionChangeInfo {
       table_moves_needed(std::move(table_moves_needed)),
       lists(std::move(lists))
     {}
+#endif
 };
 
 class DeepChangeChecker {

--- a/src/impl/collection_notifier.hpp
+++ b/src/impl/collection_notifier.hpp
@@ -52,12 +52,12 @@ struct TransactionChangeInfo {
 
     // Required by GCC 4.9
     TransactionChangeInfo() {}
-    TransactionChangeInfo(std::vector<bool> table_mods,
-                          std::vector<bool> table_moves,
-                          std::vector<ListChangeInfo> ls)
-            : table_modifications_needed(std::move(table_mods)),
-              table_moves_needed(std::move(table_moves)),
-              lists(std::move(ls))
+    TransactionChangeInfo(std::vector<bool> table_modifications_needed,
+                          std::vector<bool> table_moves_needed,
+                          std::vector<ListChangeInfo> lists)
+    : table_modifications_needed(std::move(table_modifications_needed)),
+      table_moves_needed(std::move(table_moves_needed)),
+      lists(std::move(lists))
     {}
 };
 

--- a/src/impl/collection_notifier.hpp
+++ b/src/impl/collection_notifier.hpp
@@ -55,9 +55,9 @@ struct TransactionChangeInfo {
     TransactionChangeInfo(std::vector<bool> table_mods,
                           std::vector<bool> table_moves,
                           std::vector<ListChangeInfo> ls)
-            : table_modifications_needed(table_mods),
-              table_moves_needed(table_moves),
-              lists(ls)
+            : table_modifications_needed(std::move(table_mods)),
+              table_moves_needed(std::move(table_moves)),
+              lists(std::move(ls))
     {}
 };
 

--- a/tests/util/event_loop.cpp
+++ b/tests/util/event_loop.cpp
@@ -207,7 +207,6 @@ bool EventLoop::has_implementation() { return false; }
 std::unique_ptr<EventLoop::Impl> EventLoop::Impl::main() { return nullptr; }
 EventLoop::Impl::~Impl() = default;
 void EventLoop::Impl::run_until(std::function<bool()>) { }
-void EventLoop::Impl::run_for(std::chrono::nanoseconds) { }
 void EventLoop::Impl::perform(std::function<void()>) { }
 
 #endif

--- a/tests/util/event_loop.hpp
+++ b/tests/util/event_loop.hpp
@@ -18,7 +18,6 @@
 
 #include <functional>
 #include <memory>
-#include <chrono>
 
 namespace realm {
 namespace util {

--- a/tests/util/event_loop.hpp
+++ b/tests/util/event_loop.hpp
@@ -17,6 +17,8 @@
  **************************************************************************/
 
 #include <functional>
+#include <memory>
+#include <chrono>
 
 namespace realm {
 namespace util {

